frontend/src/metabase/common/hooks/use-setting/use-setting.ts

```diff
@@ -38,5 +46,8 @@ export const useUserSetting = <T extends keyof UserSettings>(
     () => _.debounce(setter, debounceTimeout, debounceOnLeadingEdge),
     [setter, debounceTimeout, debounceOnLeadingEdge],
   );
-  return [currentValue, shouldDebounce ? debouncedSetter : setter];
+  return [
+    shouldMemoize ? memoizedValue : currentValue,
+    shouldDebounce ? debouncedSetter : setter,
```

Comment:
If this were going to go through, I think there should be a setter that does something. You should be able to at the very least have your value update to the setter's value.

Reply:
Not sure I follow

Reply:
const [value, set] = useSetting('thing', { shouldMemoize: true });
console.log(value) // 'a'

// somewhere else
set('b');

// next render
const [value, set] = useSetting('thing', { shouldMemoize: true });
console.log(value) // 'a' <-- still a even though we set it to b